tslint.json

```diff
@@ -1,3 +1,6 @@
 {
-  "extends": ["./node_modules/@stoplight/scripts/tslint.json"]
+  "extends": ["./node_modules/@stoplight/scripts/tslint.json"],
+  "rules": {
+    "no-shadowed-variable": false
```

Comment:
Can't we disable specific lines?

Reply:
Nah because it's inside the React components. I haven't found a syntax for embedding `@tsignore` in JSX. There is a ton of shadowing going on (whether it's DraftValue, etc) and now that Vincenzo has expressed a distaste for little red warnings in VS Code (regarding the JSON schema) I figured we better hide these little red warnings before he sees them. :smile:

Reply:
I also agree that this shadow variable warning is kind of a pain here and there in the codebase. I would evaluate to disable it globally in sl-scripts.

Reply:
I don't like this rule as well, but do try to follow it regardless.
You need to disable tslint, not ts :P
Example:
```tsx
    <ThrottleValue ms={1000} value={value} onChange={onChange}>
      {/* tslint:disable-next-line:no-shadowed-variable */}
      {({ value, onChange }) => 
```

Reply:
// You need to disable tslint, not ts :P
ooooooooh!!! I'll try that.

Reply:
Ah, nope. That breaks TypeScript. :-( I think because `DraftValue` expects a single child node but the comment counts as one.

```ts
<DraftValue value={key} onChange={_key => noConflict(_key) && onChange(easyObject.updateKey(index, _key))}>
  {/* tslint:disable-next-line:no-shadowed-variable */}
  {({ value, onChange }) => {
```

```
[ts]
Type '{ children: any[]; value: any; onChange: (_key: any) => false | void; }' is not assignable to type 'Readonly<IDraftValue>'.
  Types of property 'children' are incompatible.
    Type 'any[]' is not assignable to type 'FunctionComponent<IInputWidget & { nonDraftValue: any; }>'.
      Type 'any[]' provides no match for the signature '(props: IInputWidget & { nonDraftValue: any; } & { children?: ReactNode; }, context?: any): ReactElement<any> | null'. [2322]
```

Reply:
Does anyone else feel like typescript sucks beyond a certain point?

Reply:
```tsx
<DraftValue value={key} onChange={_key => noConflict(_key) && onChange(easyObject.updateKey(index, _key))}>
  {/* tslint:disable-line:no-shadowed-variable */ ({ value, onChange }) =>
```
This would do the trick I guess, but the readability is... meh